CRM/Civicase/Service/CaseMenu/Route.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+use CRM_Civicase_Service_CaseCategoryPermission as CaseCategoryPermission;
+
+/**
+ * Base class for Menu Routes.
+ *
+ * @method string getLabel()
+ * @method string getName()
+ * @method string getUrl()
+ * @method string getPermission()
+ * @method string getPermissionOperator()
+ * @method string getHasSeparator()
+ */
+abstract class CRM_Civicase_Service_CaseMenu_Route {
+
+  /**
+   * Case Type Category Name.
+   *
+   * @var string
+   */
+  protected $categoryName;
+
+  /**
+   * Case Type Category ID.
+   *
+   * @var int
+   */
+  protected $categoryId;
+
+  /**
+   * Array of permissions for the associated Category.
+   *
+   * @var array
+   */
+  protected $permissions;
+
+  /**
+   * Properties with default value overwritten.
+   *
+   * All the attributes of the route class can be overwritten here, filling
+   * this array using the appropriate setter.
+   *
+   * @var array
+   */
+  protected $overwrittenProperties;
+
+  /**
+   * Constructor of the route.
+   *
+   * @param string $categoryName
+   *   Case Type Category Name for the route.
+   * @param int $categoryId
+   *   Case Type Category Id for the route.
+   */
+  public function __construct(string $categoryName, int $categoryId) {
+    $this->permissions = (new CaseCategoryPermission())->get($categoryName);
+
+    $this->categoryName = $categoryName;
+    $this->categoryId = $categoryId;
+
+    $this->overwrittenProperties = [];
+  }
+
+  /**
```

Comment:
Its the base class, so cant be inherited doc right?

Reply:
Yes, right, I fixed it here: https://github.com/compucorp/uk.co.compucorp.civicase/pull/756/commits/16f1ef8e0a0ed641a105a908cbde8cfcacf6fe36

Reply:
Yes, I was confused here. Fixed: https://github.com/compucorp/uk.co.compucorp.civicase/pull/756/commits/66424d5a429d06bf2c4ffbd52a730bd445cc825d